### PR TITLE
Add polyfill for IE11 and fix IE/Edge layouts issues

### PR DESCRIPTION
**Pull request by popojargo**
## Overview

- Added babel-polyfill to support ES6 functions on browsers that doesn't support them
- Fixed layouts problems on IE11 and Edge

## Testing recommendations

Since we seems to mainly support Firefox and Chrome, there are not automated tests for IE.

With my changes, I fixed:
- EDGE: Fixed JSON API bar display
- IE11 : Fixed replication page that wasnt shown
- IE11 : We are now able to use React 👍 
- IE11 : Polling slider where not shown correctly (but they still don't work correctly on IE)

## GitHub issue number

#949

## Checklist

- [x] Code is written and works correctly;
- [ ] Changes are covered by tests;
- [ ] Documentation reflects the changes;
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the correct tag once a new Fauxton release is made

### CODE REVIEW

**Comment by garrensmith** [comment]
THanks @popojargo this is a good start. I'm struggling to review it though. There seems a lot of unnecessary changes in this PR. There is quite a lot of white space changes to files. Was that intentional? I can't figure out how they are related to this PR. From what I could understand with this PR the main changes are the css changes and the webpack plugin. Also can you remove the package-lock.json. The package-lock should be in a separate PR.

**Comment by popojargo** [comment]
@garrensmith  My bad with the white spaces. I am using Webstorm IDE and when I fixed the EsLint error, it automatically formatted the whole files. Do you want me to "undo" the formatting changes?

Has for the package-lock, why it's not on the repository? If we don't want to have it on the repo, shouldn't we ignore it?

**Comment by garrensmith** [comment]
@popojargo the issue here is that have a few unrelated changes all in 1 pr. The package-lock should be in the repo. But it should be in its own PR. The whitespace makes it really tricky to see what changes you have made. So its really hard to review. Ideally next time. Have all the whitespace changes in a separate commit.

File: app/addons/documents/mango/components/ExecutionStats.js [hunk 1]
```diff
@@ -122,6 +123,6 @@ export default class ExecutionStats extends React.Component {
 };
 
 ExecutionStats.propTypes = {
-  executionStats: React.PropTypes.object,
-  warning: React.PropTypes.string
+  executionStats: PropTypes.object,
```
Comment (garrensmith) [inline app/addons/documents/mango/components/ExecutionStats.js]: This is great. But unrelated to the PR

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,8938 @@
+{
+  "name": "fauxton",
```
Comment (garrensmith) [inline package-lock.json]: Lets add this in a separate PR

File: webpack.config.release.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const settings = require('./tasks/helper')
 module.exports = {
   // Entry point for static analyzer:
   entry: {
-    bundle: './app/main.js'
+    bundle: ['babel-polyfill', './app/main.js']
```
Comment (garrensmith) [inline webpack.config.release.js]: Could we narrow this down abit. What is the actuall polyfill that we are missing? I imagine its something to do with arrays.

Comment (popojargo) [reply webpack.config.release.js]: I could probably be more granular and detect which es6 functions aren't supported by IE11 and Edge. Let me look at it

**Comment by popojargo** [comment]
Closing this PR for another cleaner PR

**Comment by gdelfino** [comment]
Is this fix included in CouchDB 2.1.2 released today?

**Comment by popojargo** [comment]
@gdelfino I think so

**Comment by wohali** [comment]
CouchDB 2.1.2 has Fauxton 1.1.15, a point that missed the release notes I'm afraid.

1.1.15 includes all Fauxton improvements up through 1 May 2018.

**Pull request closed without merging**